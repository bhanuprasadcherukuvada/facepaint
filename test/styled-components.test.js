/* eslint-disable no-sparse-arrays */
import React from 'react'
import renderer from 'react-test-renderer'
import styled from 'styled-components'

import facepaint from '../src/index'

const mq = facepaint([
  '@media(min-width: 420px)',
  '@media(min-width: 920px)',
  '@media(min-width: 1120px)'
])

describe('facepaint', () => {
  test('basic', () => {
    const Div = styled('div')`
      ${mq({ color: ['red', 'green', 'blue', 'darkorchid'] })};
    `
    const tree = renderer.create(<Div>Basic</Div>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('holes', () => {
    const Div = styled('div')`
      ${mq({ color: ['red', , 'blue', 'darkorchid'] })};
    `
    const tree = renderer.create(<Div>Holes</Div>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('multiple', () => {
    const Div = styled('div')`
      ${mq({
        color: ['red', 'green', 'blue', 'darkorchid'],
        display: ['flex', 'block', 'inline-block', 'table'],
        fontSize: '12px',
        alignItems: 'center'
      })};
    `
    const tree = renderer.create(<Div>Multiple</Div>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('nested', () => {
    const Div = styled('div')`
      ${mq({
        backgroundColor: 'hotpink',
        textAlign: 'center',
        width: ['25%', '50%', '75%', '100%'],
        '& .foo': {
          color: ['red', 'green', 'blue', 'darkorchid'],
          '& img': {
            height: ['10px', '15px', '20px', '25px']
          }
        }
      })};
    `
    const tree = renderer
      .create(
        <Div>
          <div className="foo">foo</div>
          nested
        </Div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
