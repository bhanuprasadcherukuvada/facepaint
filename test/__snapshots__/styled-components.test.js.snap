// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facepaint basic 1`] = `
.c0 {
  color: red;
}

@media (min-width:420px) {
  .c0 {
    color: green;
  }
}

@media (min-width:920px) {
  .c0 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .c0 {
    color: darkorchid;
  }
}

<div
  className="c0"
>
  Basic
</div>
`;

exports[`facepaint holes 1`] = `
.c0 {
  color: red;
}

@media (min-width:920px) {
  .c0 {
    color: blue;
  }
}

@media (min-width:1120px) {
  .c0 {
    color: darkorchid;
  }
}

<div
  className="c0"
>
  Holes
</div>
`;

exports[`facepaint multiple 1`] = `
.c0 {
  color: red;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:420px) {
  .c0 {
    color: green;
    display: block;
  }
}

@media (min-width:920px) {
  .c0 {
    color: blue;
    display: inline-block;
  }
}

@media (min-width:1120px) {
  .c0 {
    color: darkorchid;
    display: table;
  }
}

<div
  className="c0"
>
  Multiple
</div>
`;

exports[`facepaint nested 1`] = `
.c0 {
  background-color: hotpink;
  text-align: center;
  width: 25%;
}

.c0 .foo {
  color: red;
}

.c0 .foo img {
  height: 10px;
}

@media (min-width:420px) {
  .c0 {
    width: 50%;
  }
}

@media (min-width:920px) {
  .c0 {
    width: 75%;
  }
}

@media (min-width:1120px) {
  .c0 {
    width: 100%;
  }
}

@media (min-width:420px) {
  .c0 .foo {
    color: green;
  }
}

@media (min-width:920px) {
  .c0 .foo {
    color: blue;
  }
}

@media (min-width:1120px) {
  .c0 .foo {
    color: darkorchid;
  }
}

@media (min-width:420px) {
  .c0 .foo img {
    height: 15px;
  }
}

@media (min-width:920px) {
  .c0 .foo img {
    height: 20px;
  }
}

@media (min-width:1120px) {
  .c0 .foo img {
    height: 25px;
  }
}

<div
  className="c0"
>
  <div
    className="foo"
  >
    foo
  </div>
  nested
</div>
`;
